.github/workflows/py-workflow.yml

```diff
@@ -0,0 +1,568 @@
+
+  Build-38:
+    if: ${{ !inputs.release && startsWith(inputs.PYTHON_VERSION, '3.8') }}
+    runs-on: ubuntu-22.04
+    needs: [ Pre-build ]
+    container:
+      image: registry.cloud.mov.ai/qa/py-buildserver:v3.0.4
+      credentials:
+        username: ${{secrets.registry_user}}
+        password: ${{secrets.registry_password}}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        submodules: recursive
+
+    - name: Upgrade pip
+      run: python -m pip install pip --upgrade
+
+    - name: Configure pip with custom index
+      run: |
+        mkdir -p ~/.config/pip
+        echo "[global]" > ~/.config/pip/pip.conf
+        echo "index-url = https://artifacts.cloud.mov.ai/repository/pypi-experimental/simple" >> ~/.config/pip/pip.conf
+        echo "extra-index-url =" >> ~/.config/pip/pip.conf
+        echo "  https://pypi.org/simple" >> ~/.config/pip/pip.conf
+        echo "trusted-host =" >> ~/.config/pip/pip.conf
+        echo "  artifacts.cloud.mov.ai" >> ~/.config/pip/pip.conf
+        echo "  pypi.org" >> ~/.config/pip/pip.conf
+
+    - name: Install build-requirements
+      run: python -m pip install -r build-requirements.txt
+
+    - name: Raise version before build
+      id: bump
+      run: |
+        python3 -m venv venv --clear
+        . venv/bin/activate
+        pip install bump-my-version==0.29.0
+        bump-my-version bump build --no-tag --no-commit
+        echo "current_version=$(bump-my-version show current_version)" | tee -a $GITHUB_OUTPUT
+
+    - name: Install python package
+      run: python -m pip install .
+
+    - name: Run Black
+      if: ${{ inputs.run_black }}
+      run: |
+        python -m black \
+          --line-length 100 \
+          --diff \
+          --check \
+          .
+
+    - name: Python Linters
+      run: |
+        python -m flake8 \
+          --isolated \
+          --exclude dist,.*\.egg-info,.git,__pycache__,.tox,venv \
+          --max-complexity 10 \
+          --max-line-length 100 \
+          --exit-zero \
+          --output-file flake8-report.txt
+
+        touch dummy-pylint-config
+
+        python -m pylint \
+          --rcfile dummy-pylint-config \
+          --ignore-patterns dist,.*\.egg-info,.git,__pycache__,.tox,venv \
+          --max-line-length 100 \
+          --exit-zero \
+          --output pylint-report.txt \
+          .
+
+    - name: Run tests
+      run: |
+        if [ -f tox.ini ]; then
+            # run tests and generates coverage.xml
+            tox
+        else
+            # run tests
+            python -m pytest tests/
+        fi
+
+    - name: Upload tests report
+      uses: actions/upload-artifact@v4
+      with:
+        name: tests-report
+        path: |
+          coverage.xml
+          flake8-report.txt
+          pylint-report.txt
+        retention-days: 5
+
+    - name: Build
+      run: |
+        python -m pip install build twine
```

Comment:
Move twine install to its step?

Reply:
I prefer to install all requirements in one step usually